### ISSUE TITLE
Limit music usage to groups

### ISSUE THREAD

**Issue by realies** [owncloud/music#604]
Would it be possible to introduce the ability to limit music usage to certain user groups?

**Comment by paulijar** [comment]
Probably it would be possible. I don't know how but I see that at least Nextcloud 12 has such ability on many applications, including e.g. the Gallery app. It might be just the matter of enabling the feature in some configuration.

**Comment by paulijar** [comment]
After some investigation it's clear that we *could* enable the 'Enable only for specific groups' functionality but not that we *should* do it. The problem is that disabling the Music app for some of the users would create new bugs related to audio file sharing. The root issue is that if Music app is disabled on a user, then it cannot react to any file system changes initiated by that user.

Consider, for example, following case: User `A` belongs to a group which can use the Music app but user `B` does not. User `B` shares an audio file with `A` but it doesn't get added to `A`'s music library automatically because the event is missed as `B` has no Music app active. When `A` opens the Music app, he is prompted about new music and can scan and add the shared file, so this is not a major issue. But if `B` now unshares or deletes the file, also that event is missed by the Music app. Hence, `A` is left with a dangling reference to a file which is no longer accessible. `A` has no easy way to get rid of this ghost file so it may stay in his music library forever; apart from using the `occ` command line utility, the only way would be to change the music library folder from the personal settings to enforce a complete rescan.

For very similar reason, the feature is disabled on the Activity app, see https://github.com/nextcloud/server/issues/3707. In the comments of that issue, it's said that improved framework-side support would be needed before enabling the feature could be considered. If that ever happens, then we can re-evaluate it also on the Music app. But for now, I would say it's not worth the trouble.

Anyway, if you really need this feature on your installation, it's rather simple to enable it. Just modify the file `apps/music/appinfo/info.xml` and completely remove the block
```
	<types>
		<!-- update metadata cache when create/update/delete a file -->
		<filesystem/>
	</types>
```
and it should work.

**Comment by BuzencoA** [comment]
It is still not possible?

**Comment by paulijar** [comment]
@BuzencoA Yep, it's still not possible. All we are telling to the ownCloud/Nexcloud core is that the Music app wants to observe the filesystem events. Based on this information, the core disables the ability to limit the app to groups.

I believe that if the core side support was implemented as discussed in nextcloud/server#3707, then this feature would automatically become available also for the Music app. But based on the last comment on that issue, there are no plans to implement this in the core.
